### ISSUE TITLE
Robotframework wheel creation results in only the scripts for the platform being copied on installation

### ISSUE THREAD

**Issue by michaelusner** [robotframework/robotframework#2983]
## Description
After building a wheel (pip wheel -w dist .) we end up with an apparent platform independent file:
robotframework-3.0.4-py2-none-any.whl

When installing this wheel on a different platform, the correct scripts (pybot, robot, rebot) are not copied into the Scripts folder.  Only the platform scripts for the build machine are copied.

- Wheels built in Linux won't have scripts for Windows (robot.bat, rebot.bat, pybot.bat)
- Wheels built in Windows won't have scripts for Linux (robot, rebot, pybot)

## Possible fixes:

- Building the wheel copies scripts for both platforms
- Builds can produce platform explicit wheels (not optimal)

## Version information
Robot Framework version: 3.0.4
Python 2.7.15
Operating system and its version
Windows (8/10/2016) / Linux (CentOS7)

### Linux
  - Build wheel
    ```
      pip wheel -w dist .
      unzip -l dist/robotframework-3.0.4-py2-none-any.whl | grep scripts
       70  2018-10-12 17:55   robotframework-3.0.4.data/scripts/pybot
      192  2018-10-12 17:55   robotframework-3.0.4.data/scripts/rebot
      188  2018-10-12 17:55   robotframework-3.0.4.data/scripts/robot
    ```
    Note that the respective bat files are not present, so this build won't work in Windows without an explicit "python -m robot.run".

### Windows
  - Build wheel
    ```
    pip wheel -w dist .

    unzip -l dist\robotframework-3.0.4-py2-none-any.whl | find "scripts"
     35  2018-09-06 16:43   robotframework-3.0.4.data/scripts/pybot.bat
     90  2018-10-12 18:33   robotframework-3.0.4.data/scripts/rebot.bat
     88  2018-10-12 18:33   robotframework-3.0.4.data/scripts/robot.bat
    ``` 
    Note that the respective shell scripts aren't present so this build won't work in Linux without an explicit "python -m robot.run".

**Comment by pekkaklarck** [comment]
Please test RF 3.1 preview releases. They use wheel by default, start-up scripts are exe-files on Windows, etc.

https://github.com/robotframework/robotframework/blob/master/doc/releasenotes/rf-3.1a2.rst

**Comment by michaelusner** [comment]
Thanks, I did see that the setup had changed in the preview version but
since we're using RF in production testing, I was reluctant to try it.

Thanks for the response.

**Comment by pekkaklarck** [comment]
It would be good if you could test the 3.1 preview releases to see how installation enhancements work in your environment. There's not going to be any more 3.0.x releases, and getting possible problems fixed before 3.1 final would be obviously be great.

**Comment by pekkaklarck** [comment]
This issue is basically a duplicate of #1734 that is now fixed will be part of RF 3.1. The reason we couldn't provide wheels earlier was that our setup.py operating system specific code what wasn't executed when using wheels. That's also the exact reason for the problems mentioned in this issue and also caused problems explained in #2529. The problems preventing wheel usage were removed by #2415 and #2818. All the problems mentioned in this issue ought to thus be fixed now. If there are still problems, new issues should be submitted about them.

**Comment by michaelusner** [comment]
Confirmed fixed in 3.1a2.  Thanks for your help!
